batchMTE_test.go

```
@@ -254,25 +225,17 @@ func TestBatchMTEIdentificationNumber(t *testing.T) {
 
 	// NACHA rules state MTE records can't be all spaces or all zeros
 	mockBatch.GetEntries()[0].IdentificationNumber = "   "
-	if err := mockBatch.Create(); err != nil {
-		if e, ok := err.(*BatchError); ok {
-			if e.FieldName != "IdentificationNumber" {
-				t.Errorf("%T: %s", err, err)
-			}
-		} else {
-			t.Errorf("%T: %s", err, err)
-		}
+	err := mockBatch.Create()
+	// TODO: are we expecting there to be no errors here?
+	if !base.Match(err, nil) {
+		t.Errorf("%T: %s", err, err)
 	}
 
 	mockBatch.GetEntries()[0].IdentificationNumber = "000000"
-	if err := mockBatch.Create(); err != nil {
-		if e, ok := err.(*BatchError); ok {
-			if e.FieldName != "IdentificationNumber" {
-				t.Errorf("%T: %s", err, err)
-			}
-		} else {
-			t.Errorf("%T: %s", err, err)
-		}
+	err = mockBatch.Create()
+	// TODO: are we expecting there to be no errors here?
```

Yes, we should return an error for each case when ranging through entries in batchMTE

for _, entry := range batch.Entries {...